components/org.wso2.carbon.identity.local.auth.smsotp.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/smsotp/authenticator/SMSOTPAuthenticator.java

```diff
@@ -612,10 +612,14 @@ private UserStoreManager getUserStoreManager(AuthenticatedUser authenticatedUser
     private String resolveMobileNoAttribute(AuthenticatedUser user, String tenantDomain,
                                             AuthenticationContext context) throws AuthenticationFailedException {
 
+        // Prioritizing the authenticator's dialect first, then considering the claim mapping defined in the IdP.
         String dialect = getFederatedAuthenticatorDialect(context);
         if (SMSOTPConstants.OIDC_DIALECT_URI.equals(dialect)) {
             return SMSOTPConstants.MOBILE_ATTRIBUTE_KEY;
         }
+        if (SMSOTPConstants.WSO2_CLAIM_DIALECT.equals(dialect)) {
+            return SMSOTPConstants.MOBILE_ATTRIBUTE_KEY;
+        }
```

Comment:
This does not return the full claim. Is  the logic handled to construct the claim ? Looks not based on line 617 and 618